cnd/src/ethereum/mod.rs

```diff
@@ -1,11 +1,78 @@
 #![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
 #![forbid(unsafe_code)]
 
-pub use web3::types::{
-    Address, Block, BlockId, BlockNumber, Bytes, Log, Transaction, TransactionReceipt,
-    TransactionRequest, H160, H2048, H256, U128, U256,
+pub use primitive_types::{
+    H160, H256, U128, U256
 };
 
+pub use ethbloom::{Bloom as H2048, Input};
+
+use fixed_hash::*;
+use impl_rlp::impl_fixed_hash_rlp;
+use impl_serde::impl_fixed_hash_serde;
+
+use impl_rlp::impl_uint_rlp;
+use impl_serde::impl_uint_serde;
+use uint::*;
+pub use uint::FromDecStrErr;
+
+use rustc_hex::{FromHex, ToHex};
+use serde::de::{Error, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt;
+
+pub type Index = U128;
+pub type Address = H160;
+
+construct_fixed_hash! {
+    pub struct H64(8);
+}
+impl_fixed_hash_rlp!(H64, 8);
```

Comment:
I don't think we need the rlp encoding as we are never deserializing/serializing to the binary format of Ethereum transactions.

Reply:
Would be nice to avoid the `impl_rlp` dependency.